xonsh/parsers/base.py

```diff
@@ -1973,9 +1977,15 @@ def p_nocloser_many(self, p):
     def p_string_literal(self, p):
         """string_literal : string_tok"""
         p1 = p[1]
-        s = ast.literal_eval(p1.value)
-        cls = ast.Bytes if p1.value.startswith('b') else ast.Str
-        p[0] = cls(s=s, lineno=p1.lineno, col_offset=p1.lexpos)
+        if p1.value.startswith('p'):
+            s = ast.Str(s=ast.literal_eval(p1.value[1:]), lineno=p1.lineno,
+                        col_offset=p1.lexpos)
+            p[0] = xonsh_call('__xonsh_path_literal__', [s],
+                              lineno=p1.lineno, col=p1.lexpos)
```

Comment:
How would you prefer to go about this?

It is necessary to have some logic at this point since `p` is part of the string token but `ast.literal_eval` can't handle it. You could perhaps introduce a custom AST node and generate the path later, but that seems like it would introduce more complexity.